py/objdeque.c

```diff
@@ -135,6 +183,74 @@ STATIC mp_obj_t deque_popleft(mp_obj_t self_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(deque_popleft_obj, deque_popleft);
 
+STATIC mp_obj_t deque_pop(mp_obj_t self_in) {
+    mp_obj_deque_t *self = MP_OBJ_TO_PTR(self_in);
+
+    if (self->i_get == self->i_put) {
+        mp_raise_msg(&mp_type_IndexError, MP_ERROR_TEXT("empty"));
+    }
+
+    if (self->i_put == 0) {
+        self->i_put = self->alloc - 1;
+    } else {
+        self->i_put--;
+    }
+
+    mp_obj_t ret = self->items[self->i_put];
+
+    return ret;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(deque_pop_obj, deque_pop);
+
+STATIC mp_obj_t deque_getiter(mp_obj_t o_in, mp_obj_iter_buf_t *iter_buf) {
+    mp_obj_deque_t *deque = MP_OBJ_TO_PTR(o_in);
+    return mp_obj_new_deque_iterator(o_in, deque->i_get, iter_buf);
+}
+
+STATIC void mp_obj_deque_store(mp_obj_t self_in, mp_obj_t index, mp_obj_t value) {
+    mp_obj_deque_t *self = MP_OBJ_TO_PTR(self_in);
+    size_t offset = mp_get_index(self->base.type, deque_len(self), index, false);
+    size_t i = (self->i_get + offset) % self->alloc;
+    self->items[i] = value;
+}
+
+STATIC mp_obj_t deque_subscr(mp_obj_t self_in, mp_obj_t index, mp_obj_t value) {
+    if (value == MP_OBJ_NULL) {
+        // delete not supported, but must match CPython error message
```

Comment:
For unimplemented features it's not necessary to match CPython with the error type/message.  And I think `mp_obj_subscr()` already gives decent error messages for the default cases when these branches here are not implemented.  So I think it's possible to just do nothing in this delete case.

Reply:
Ah, I see that now. That's a neat trick, falling through to the raise statements if you return `MP_OBJ_NULL`. IMO, a comment to that effect would be helpful to remind people of that behavior.

Reply:
There is a (short) comment about this here: https://github.com/micropython/micropython/blob/master/py/obj.h#L644